homeassistant/auth/permissions/entities.py

```diff
@@ -77,6 +76,41 @@ def _lookup_area(
     return area_dict.get(device_entry.area_id)
 
 
+def _lookup_labels(
+    perm_lookup: PermissionLookup, label_dict: SubCategoryDict, entity_id: str
+) -> ValueType | None:
+    """Look up entity permissions by label."""
+    entity_entry = perm_lookup.entity_registry.async_get(entity_id)
+
+    if entity_entry is None:
+        return None
+    if not entity_entry.labels:
+        return None
+
+    policy: Mapping[str, bool] = {}
+
+    for label_id in entity_entry.labels:
```

Comment:
This lookup is very slow, which impacts the performance of our auth system significantly. Since it's not currently being used inside Home Assistant, I don't feel comfortable adding this feature at this time.

Reply:
As far as I understand this loop will only be executed if the security by label is used, wouldn't it be enough with warning the user that security by label is slow? Other users will not be affected at all.

If this is not acceptable, do you have any suggestions about how to make it faster? 

Thanks!